lib/server.js

```diff
@@ -322,9 +322,15 @@ var SolidusServer = function( options ){
 
   // updates the source of a preprocessor
   this.updatePreprocessor = function( preprocessor_path ){
+    var preprocessor_regex = new RegExp( escapePathForRegex( paths.preprocessors ) +'.+\.js', 'i' );
 
-    delete require.cache[require.resolve(preprocessor_path)];
+    var preprocessor_paths = Object.keys(require.cache).filter(function( path ){
+      return preprocessor_regex.test(path);
+    });
```

Comment:
Oh! I missed that there was a var with them all in this scope, despite the method immediately above adding to it